Libraries/LibWeb/Painting/ViewportPaintable.cpp

```diff
@@ -374,7 +374,8 @@ void ViewportPaintable::recompute_selection_states(DOM::Range& range)
     }
 
     // 5. Mark the nodes between start node and end node (in tree order) as Full.
-    for (auto* node = start_container->next_in_pre_order(); node && (node->is_before(end_container) || node->is_descendant_of(end_container)); node = node->next_in_pre_order()) {
+    auto* stop = end_container->next_in_pre_order();
+    for (auto* node = start_container->next_in_pre_order(); node && node != stop; node = node->next_in_pre_order()) {
```

Comment:
Is it guaranteed that `start_container` is always before `end_container`? If not this behaves differently than before.

Reply:
good question, I just checked and all `m_start_container` and `m_end_container` modifications seem to be going through `Range::set_start_or_end()`, which takes care of maintaining the invariant that start comes before end, so it should be safe to make this assumption here.